### ISSUE TITLE
 Kasm Workspace Please

### ISSUE THREAD

**Issue by nmcbride** [xpipe-io/xpipe-webtop#3]
Since you are using the Kasm VNC container, could you please stand up a xpipe kasm workspace repository so we can easily add it to our kasm servers?

Thank you.

**Comment by crschnick** [comment]
I can look into it. I have not used that before, so I don't know how much work that is

**Comment by crschnick** [comment]
Can you try whether this works for you? https://github.com/xpipe-io/kasm-registry

I can't test it because my Kasm Workspaces selfhosted setup doesn't work at all with any image. Everything is completely broken for me

**Comment by nmcbride** [comment]
@crschnick 

Thank you for working on this. I tried adding it.

I initially got an error but then the registry added. But now it doesn't show anything. It doesn't show the other registries I had added and if I try to readd them it says they are already added and it doesn't show any workspaces available to be installed. This of course includes the xpipe one.

I'm assuming maybe this is also what happened to you when you tried to add it?

If I find a way to fix my server I'll let you know so we can continue working on it. Otherwise I may have to rebuild.

**Comment by nmcbride** [comment]
@crschnick To fix my server I did;

- stopped everything
- modified their docker-compose.yml to have the DB bind to a host port
- modified their pg_hba.conf to allow their user to connect in from any ip address
  - i could not get the password to work for some reason so I set it to Trust 
  - ssl connections will be forced to password auth due to a setting in their pg_hba.conf
- spun up pgadmin4 in a container on the host network
- added a connection to the host ip for user kasmapp and turned the ssl setting to disable
- went to the registry table under kasm -> public -> tables -> registries
- deleted the xpipe one

Everything is back to normal and working again.

I do not know what it is yet but I'm assuming perhaps there is something in the xpipe registry config that was left the default value and is conflicting with the default kasm repository that is added on install.

I am going to snapshot this VM before I play further.

At a quick glance I do not see anything that is the default, but I think something is not right atm.

**Comment by crschnick** [comment]
Yeah I had exactly the same issue. It is almost impossible to debug because the logs don't say anything.

I double checked the registry config and fixed a wrong registry URL. Maybe that was the issue?

**Comment by crschnick** [comment]
I managed to get the registry working. You can try it out.

The issue now is that the workspace starts up to a black screen. And I am out of things to try. If you want, you can try to figure out why the image, which is very close to the linuxserver webtop images, doesn't work.

You can compare the images at https://github.com/xpipe-io/xpipe-webtop/blob/main/Dockerfile and https://github.com/linuxserver/docker-webtop/blob/ubuntu-kde/Dockerfile

You can compare the registry entries at https://github.com/xpipe-io/kasm-registry/tree/1.1/workspaces/xpipe-webtop and https://github.com/linuxserver/kasm_workspaces_registry/tree/1.1/workspaces/ubuntu-kde,

**Comment by nmcbride** [comment]
@crschnick What was the issue with the registry?

**Comment by crschnick** [comment]
The problem with the registry was that kasm now uses schema version 1.1 but all the tutorials covered only 1.0, so I had the wrong schema.

**Comment by crschnick** [comment]
I'm now marking this issue as help wanted as I am out of my depth here.

If anyone can figure out why the xpipe workspace only produces a black screen while the very similar ubuntu kde workspace works, that would be appreciated. See https://github.com/xpipe-io/xpipe-webtop/issues/3#issuecomment-2564587515

**Comment by nmcbride** [comment]
Myself and a friend will be looking at it next week but we are both away from our main rigs for a week.

**Comment by crschnick** [comment]
Thanks, appreciate the help

**Comment by nmcbride** [comment]
@crschnick 

My friend said adding this to docker exec config sovled the issue for him. I tested and it also worked for me so seems like a simple permission issue:
```
{
  "first_launch": {
    "cmd": "bash -c 'sudo chmod +x /home/kasm-user/.vnc/xstartup'"
  }
}
```

I'd also do a few other things like:
 - Just call it "XPipe" not "XPipe Webtop"
 - Configure the docker image to start it on start in a maximized window

**Comment by nmcbride** [comment]
My friend also tried it for the first time and tried SSHing into one of his rPi 4/5s and it said you could not do that on the free license. Is that on purpose for a rPi? or is it triggering as something else like an embedded system or something like that?

**Comment by crschnick** [comment]
Alright, thanks for the help. I fixed the original image with the executable permission issue.

I tested it on my setup and it works now. I also fixed the automatic XPipe startup on launch, but I'm not sure whether it applies. I don't know how to repull an updated image in kasm. About the name, I think if I just call it XPipe, it might get confused for these single applications that are also supported by Kasm. I want to indicate that it is a whole desktop environment.

The rPi license issue should be solved, it was about the disabled system interaction toggle that was on even though it was not needed. It is supported in the community version.
![image](https://github.com/user-attachments/assets/1bca87cb-1a40-43e5-bca9-acf03646a90d)

**Comment by crschnick** [comment]
So feel free to test it. If everything works, I can even squeeze it into the XPipe 14 release announcement

**Comment by nmcbride** [comment]
I'll let him know thank you.

**Comment by nmcbride** [comment]
turning that off does seem to allow it to work. ty.

**Comment by crschnick** [comment]
I think I can close this issue now. Thanks for the help of you guys.

XPipe 14 is also now available in the webtop
